src/test/java/org/mockbukkit/mockbukkit/entity/ai/BrainMockTest.java

```diff
@@ -0,0 +1,198 @@
+package org.mockbukkit.mockbukkit.entity.ai;
+
+import org.bukkit.entity.memory.MemoryKey;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+class BrainMockTest
+{
+	private final BrainMock brain = new BrainMock();
+
+	@Nested
+	class SetMemory
```

Comment:
It should also be possible to set a key to null, if the value is null, it looks like it clears the existing value from the memory map.

Reply:
This was to achieve the same behavior as Minecraft, but attempting to make it a "clean" approach.
At the moment Minecraft "brain" uses an Optional value. So you can have:
- Pair does not exist in the map.
- Pair has an empty optional as value
- Pair has a value

I tried to keep the same behavior but without using the optional:
- Pair does not exist in the map.
- Pair has a null value
- Pair has a value

Reply:
Mmm, okay. But inserting a null value for the value parameter in `LivingEntityMock#setMemory` should clear the memory for that key.